src/org/mozilla/javascript/Parser.java

```diff
@@ -233,8 +233,12 @@ void addError(String messageId, int position, int length) {
     }
 
     void addError(String messageId, String messageArg) {
-        addError(messageId, messageArg, ts.tokenBeg,
-                 ts.tokenEnd - ts.tokenBeg);
+        int beg = -1, end = -1;
```

Comment:
I'd really like us to adopt a single-assignment style for local variables in Rhino whenever possible, so
```java
final int beg, end;
if (ts != null) {
    ...
} else {
    beg = end = -1;
}
```
would be much preferred.